### PR TITLE
feat: GCP cost collection

### PR DESCRIPTION
**Pull request by Jelledb**
(I still have to write the config part of this code, it's just hardcoded for now, but it works)

Opening this already for visibility

### CODE REVIEW

File: src/config/gcp-post-config.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    });
+
+    await this.createMaterializedView(projectId, datasetName, tableName);
+
+    config.gcp.billingExport = {
+      projectId,
+      datasetName,
+      tableName,
+    };
+
```
Comment (Jelledb) [inline src/config/gcp-post-config.ts:65]: self: add successful message that user is now ready to use GCP cost collection

File: src/config/gcp-post-config.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  async executeDisconnected(_config: Config) {
+    // no op
+  }
+
+  private async listDatasets(projectId: string): Promise<string[]> {
+    const result = await this.shellRunner.run(
+      `bq ls --project_id ${projectId} --format json`,
+    );
+    console.log(result);
```
Comment (Jelledb) [inline src/config/gcp-post-config.ts:77]: remove console

File: src/config/post-config-hooks.ts [hunk 3]
```diff
@@ -1,9 +1,11 @@
 import { ShellRunner } from "../process/shell-runner.ts";
 import { AwsPostPlatformConfigHook } from "./aws-post-config.ts";
 import { PostPlatformConfigHook } from "./post-platform-config-hook.ts";
+import { GcpPostPlatformConfigHook } from "./gcp-post-config.ts";
 
 export function buildConfigHooks(): PostPlatformConfigHook[] {
   return [
     new AwsPostPlatformConfigHook(new ShellRunner()),
+    new GcpPostPlatformConfigHook(new ShellRunner()),
```
Comment (Jelledb) [inline src/config/post-config-hooks.ts:9]: @tfelix maybe we should share the shellrunner here? or what do you think?

**Review by tfelix — APPROVED** [review]

File: src/config/config.model.ts [hunk 4]
```diff
@@ -65,6 +74,7 @@ export const emptyConfig: Config = {
     parentManagementGroups: [],
   },
   aws: {},
+  gcp: {},
```
[the remark this answers is not in the document]
Comment (Jelledb) [reply src/config/config.model.ts:77]: Good remark, will integrate something for that

File: src/config/gcp-post-config.ts [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  constructor(
+    private readonly shellRunner: ShellRunner,
+  ) {}
+
+  isExecutable(platform: ConnectedConfigKey): boolean {
+    return platform === "GCP";
+  }
+
+  async executeConnected(config: Config) {
+    if (config.gcp.billingExport) {
```
[the remark this answers is not in the document]
Comment (Jelledb) [reply src/config/gcp-post-config.ts:22]: Not sure what you mean exactly here: do you mean the scenario where you `--connect`, then `--disconnect`, and then `--connect` yet again?

**Pull request merged**